miniconf/src/postcard.rs

```diff
@@ -0,0 +1,76 @@
+use crate::{Error, IntoKeys, TreeDeserialize, TreeSerialize};
+use postcard::{de_flavors, ser_flavors, Deserializer, Serializer};
+
+/// Miniconf with `postcard`.
+pub trait Postcard<'de, const Y: usize = 1>: TreeSerialize<Y> + TreeDeserialize<'de, Y> {
+    /// Deserialize and set a node value from a `postcard` flavor.
+    ///
+    /// ```
+    /// # use miniconf::{Tree, TreeKey, Postcard, Packed};
+    /// # use postcard::de_flavors::Slice;
+    /// #[derive(Tree, Default)]
+    /// struct S {
+    ///     foo: u32,
+    ///     #[tree(depth=1)]
+    ///     bar: [u16; 3],
+    /// };
+    /// let mut s = S::default();
+    /// let p = Packed::from_lsb(0b1110.try_into().unwrap());
```

Comment:
I'm not following this. I would expect:
1. The leading `0b1, indicates leaf tree `bar`
2. The remaining `0b110` only needs 2 bits to index all of `bar` (since there are 3 elements, 2 bits should be required)
3. `0b11` should be the index into `bar`, which would be invalid, since `S::bar[3]` is an out-of-index error.

Maybe we should explain the packed key a bit more? It's hard to wrap my head around.

Also, what's the intended use case here? I don't imagine people will actually be manually calculating these packed bitfields.

Reply:
Docs for `Packed` are:

https://github.com/quartiq/miniconf/blob/f04a851878968107df92abe6e866538b3eb65547/miniconf/src/packed.rs#L7-L30

I'll spell out the example.

Reply:
Done. I'll spell out a use case in the Packed struct doc.